src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -446,6 +446,7 @@ class WhisperPreTrainedModel(PreTrainedModel):
     base_model_prefix = "model"
     main_input_name = "input_features"
     supports_gradient_checkpointing = True
+    _no_split_modules = ["WhisperEncoderLayer"]
```

Comment:
Hum I think this won't work since `WhisperEncoderLayer` has a skip connection with a residual `hidden_states`, We can fix it by forcing [this](https://github.com/huggingface/transformers/blob/072dfdaee4deee65328071e92ac48f471a78490c/src/transformers/models/whisper/modeling_whisper.py#L293) variable to be on the same device than the `hidden_states` after the attention layer but I wanted to avoid extra device moving!